staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -423,20 +398,293 @@ func TestPriorityAndFairnessWithPanicRecoverAndTimeoutFilter(t *testing.T) {
 			t.Errorf("expected inner handler to be executed for request: %s", secondRequestPathShouldWork)
 		}
 		if err != nil {
-			t.Errorf("expected request: %s to succeed, but got error: %#v", secondRequestPathShouldWork, err)
+			t.Fatalf("expected request: %s to succeed, but got error: %#v", secondRequestPathShouldWork, err)
 		}
 		if response.StatusCode != http.StatusOK {
 			t.Errorf("expected HTTP status code: %d for request: %s, but got: %#v", http.StatusOK, secondRequestPathShouldWork, response)
 		}
 
 		close(stopCh)
 		t.Log("waiting for the controller to shutdown")
-		<-controllerCompletedCh
 
+		controllerErr := <-controllerCompletedCh
 		if controllerErr != nil {
 			t.Errorf("expected a nil error from controller, but got: %#v", controllerErr)
 		}
 	})
+
+	t.Run("priority level concurrency is set to 1, request times out and inner handler hasn't written to the response yet", func(t *testing.T) {
+		const (
+			requestTimeout                                        = 1 * time.Second
+			userName                                              = "alice"
+			fsName                                                = "test-fs"
+			plName                                                = "test-pl"
+			serverConcurrency, plConcurrencyShares, plConcurrency = 1, 1, 1
+		)
+
+		apfConfiguration := newConfiguration(fsName, plName, userName, plConcurrencyShares, 0)
+		stopCh := make(chan struct{})
+		controller, controllerCompletedCh := startAPFController(t, stopCh, apfConfiguration, serverConcurrency, requestTimeout/4, plName, plConcurrency)
+
+		var executed bool
+		rquestTimesOutPath := "/request/time-out"
+		ctx, cancel := context.WithTimeout(context.Background(), requestTimeout+time.Second)
```

Comment:
I see that the context created here is not the context used in the handler chain.  That's a bit confusing.  The context created here is just used for the `<-ctx.Done()` statement, right?  I think it would be less confusing to use the `time` package directly.

Reply:
I have removed timeout using `context` and using simple channel driven sync to eliminate flakes.